examples/asset/temp_asset.rs

```diff
@@ -0,0 +1,183 @@
+        .run();
+}
+
+/// Attempt to save an asset to the temporary asset source.
+fn save_temp_asset(assets: Res<AssetServer>, mut commands: Commands) {
+    // This is the asset we will attempt to save.
+    let my_text_asset = TextAsset("Hello World!".to_owned());
+
+    // To ensure the `Task` can outlive this function, we must provide owned versions
+    // of the `AssetServer` and our desired path.
+    let path = AssetPath::from("temp://message.txt").into_owned();
+    let server = assets.clone();
+
+    let task = IoTaskPool::get().spawn(async move {
+        save_asset(my_text_asset, path, server, TextSaver)
+            .await
+            .unwrap();
+    });
+
+    // To ensure the task completes before we try loading, we will manually poll this task
+    // so we can react to its completion.
+    commands.spawn(SavingTask(task));
+}
+
+/// Poll the save tasks until completion, and then start loading our temporary text asset.
+fn wait_until_temp_saved(
+    assets: Res<AssetServer>,
+    mut tasks: Query<(Entity, &mut SavingTask)>,
+    mut commands: Commands,
+) {
+    for (entity, mut task) in tasks.iter_mut() {
+        if let Some(()) = block_on(future::poll_once(&mut task.0)) {
+            commands.insert_resource(MyTempText {
+                text: assets.load("temp://message.txt"),
+            });
+
+            commands.entity(entity).despawn_recursive();
+        }
+    }
+}
+
+/// Setup a basic UI to display our [`TextAsset`] once it's loaded.
+fn setup_ui(mut commands: Commands) {
+    commands.spawn(Camera2dBundle::default());
+
+    commands.spawn((TextBundle::from_section("Loading...", default())
+        .with_text_justify(JustifyText::Center)
+        .with_style(Style {
+            position_type: PositionType::Absolute,
+            bottom: Val::Percent(50.),
+            right: Val::Percent(50.),
+            ..default()
+        }),));
+}
+
+/// Once the [`TextAsset`] is loaded, update our display text to its contents.
+fn display_text(
+    mut query: Query<&mut Text>,
+    my_text: Option<Res<MyTempText>>,
+    texts: Res<Assets<TextAsset>>,
+) {
+    let message = my_text
+        .as_ref()
+        .and_then(|resource| texts.get(&resource.text))
+        .map(|text| text.0.as_str())
+        .unwrap_or("Loading...");
+
+    for mut text in query.iter_mut() {
+        *text = Text::from_section(message, default());
+    }
+}
+
+/// Save an [`Asset`] at the provided path. Returns [`None`] on failure.
+async fn save_asset<A: Asset>(
+    asset: A,
+    path: AssetPath<'_>,
+    server: AssetServer,
+    saver: impl AssetSaver<Asset = A> + ErasedAssetSaver,
+) -> Option<()> {
+    let asset = ErasedLoadedAsset::from(LoadedAsset::from(asset));
+    let source = server.get_source(path.source()).ok()?;
+    let writer = source.writer().ok()?;
+
+    let mut writer = writer.write(path.path()).await.ok()?;
+    ErasedAssetSaver::save(&saver, &mut writer, &asset, &())
+        .await
+        .ok()?;
+
+    Some(())
+}
```

Comment:
I _don't_ like this. Obviously the error "handling" is sub-par, and it feels like this entire function is such a mechanical task with Bevy that it should be a provided method. If we're adding a `temp://` source, we're encouraging people to save assets, so the whole process should be at least mostly as clean as loading an asset already is.